tfjs-backend-webgpu/src/reverse_webgpu.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {getMainHeaderString as main, WebGPUProgram} from './webgpu_program';
+import {computeDispatch, flatDispatchLayout} from './webgpu_util';
+
+export class ReverseProgram implements WebGPUProgram {
+  outputShape: number[];
+  shaderKey: string;
+  dispatchLayout: {x: number[]};
+  dispatch: [number, number, number];
+  variableNames = ['x'];
+  uniforms: string;
+  workgroupSize: [number, number, number] = [64, 1, 1];
+  size = true;
+
+  constructor(xShape: [number, number, number, number]) {
+    this.outputShape = xShape;
+    this.dispatchLayout = flatDispatchLayout(this.outputShape);
+    this.dispatch = computeDispatch(
+        this.dispatchLayout, this.outputShape, this.workgroupSize);
+    this.uniforms = ` axis : vec4<i32>,`;
+    this.shaderKey = 'reverse';
+  }
+
+  getUserCode(): string {
+    const inputCoordsSnippet = `
+      fn getReverseInputCoords(coords : vec4<i32>) -> vec4<i32> {
+        var inputCoords: vec4<i32>;
+        if ((uniforms.axis[0] == 1) && (uniforms.xShape[0] != 1)) {
```

Comment:
From performance point, I'd prefer you don't make axis as a uniform. And  meanwhile squeeze `x` so that there is no `1` in shape. In this case, we can avoid all `if` here?

Reply:
Or we can still keep axis as a uniform, but update the value in it so that we can always use `inputCoords[i] = uniforms.xShape[1] - coords[1] - 1`?

Reply:
When we launching a kernel, these values of `uniforms.axis[0...3] == 1` and `uniforms.xShape[0...3] != 1` are the same for all threads, all threads enter into either `if block` or `else block`. There is no situation that some threads enter `if block`, and others enter `else block`. So I think there is no performance issue. Do you think so?

Reply:
Thanks for the explanation. Please add this comment into the code in case someone revisits this code in future.

Reply:
Done